--- ./console/cake.php	2011-03-31 14:03:30.000000000 +0900
+++ /home/shwada/works/apps/works.nanapi.jp/cake/console/cake.php	2011-03-31 09:36:20.000000000 +0900
@@ -491,7 +491,6 @@
 	function parseParams($params) {
 		$this->__parseParams($params);
 		$defaults = array('app' => 'app', 'root' => dirname(dirname(dirname(__FILE__))), 'working' => null, 'webroot' => 'webroot');
-
 		$params = array_merge($defaults, array_intersect_key($this->params, $defaults));
 		$isWin = false;
 		foreach ($defaults as $default => $value) {
@@ -502,9 +501,6 @@
 		}
 		$params = str_replace('\\', '/', $params);
 
-		if (isset($params['working'])) {
-			$params['working'] = trim($params['working']);
-		}
 		if (!empty($params['working']) && (!isset($this->args[0]) || isset($this->args[0]) && $this->args[0]{0} !== '.')) {
 			if (empty($this->params['app']) && $params['working'] != $params['root']) {
 				$params['root'] = dirname($params['working']);
@@ -519,11 +515,9 @@
 		} elseif (strpos($params['app'], '/')) {
 			$params['root'] .= '/' . dirname($params['app']);
 		}
-		
-		$params['working'] = rtrim($params['root'], '/');
-		if (!$isWin || !preg_match('/^[A-Z]:$/i', $params['app'])) {
-			$params['working'] .= '/' . $params['app'];
-		}
+
+		$params['app'] = basename($params['app']);
+		$params['working'] = rtrim($params['root'], '/') . '/' . $params['app'];
 
 		if (!empty($matches[0]) || !empty($isWin)) {
 			$params = str_replace('/', '\\', $params);
